color/src/parse.rs

```diff
@@ -3,19 +3,95 @@
 
 //! Parse CSS4 color
 
+use core::error::Error;
 use core::f64;
+use core::fmt;
 use core::str::FromStr;
 
 use crate::{AlphaColor, ColorSpaceTag, DynamicColor, Missing, Srgb};
 
-// TODO: proper error type, maybe include string offset
+// TODO: maybe include string offset
 /// Error type for parse errors.
 ///
-/// Currently just a static string, but likely will be changed to
-/// an enum.
-///
 /// Discussion question: should it also contain a string offset?
-pub type Error = &'static str;
+#[derive(Clone, Debug, Eq, PartialEq)]
+#[non_exhaustive]
+pub enum ParseError {
+    /// Unclosed comment
+    UnclosedComment,
+    /// Unknown angle dimension
+    UnknownAngleDimension,
+    /// Unknown angle
+    UnknownAngle,
+    /// Unknown color component
+    UnknownColorComponent,
+    /// Unknown color identifier
+    UnknownColorIdentifier,
+    /// Unknown color space
+    UnknownColorSpace,
+    /// Unknown color syntax
+    UnknownColorSyntax,
+    /// Expected arguments
+    ExpectedArguments,
+    /// Expected closing parenthesis
+    ExpectedClosingParenthesis,
+    /// Expected color space identifier
+    ExpectedColorSpaceIdentifier,
+    /// Expected comma
+    ExpectedComma,
+    /// Invalid hex digit
+    InvalidHexDigit,
+    /// Wrong number of hex digits
+    WrongNumberOfHexDigits,
+}
+
+impl Error for ParseError {}
+
+impl fmt::Display for ParseError {
```

Comment:
In general, I like to see the types first, then the impls, but not sure about that here. If you want to move it into a separate file, it'd be fine (and wouldn't require another approval).